controller/users.controller.js

```diff
@@ -197,32 +233,89 @@ const updateUserSubscriptionStatus = async (req, res, next) => {
     const { subscription } = value;
 
     if (error) {
-      return res.status(400).json({ message: error.message });
+      return res.status(400).json({ status: 'error', code: 400, message: error.message });
     }
 
     const userId = req.user.id;
 
     await updateKeyInDBForUserWithId({ subscription }, userId);
     return res.json({
-      status: 'Success',
+      status: 'success',
       code: 200,
       message: `User's subscription changed successfully to ${subscription}.`,
     });
   } catch (err) {
     console.error(err);
     return res.status(500).json({
-      status: 'Error',
+      status: 'error',
+      code: 500,
+      message: 'Server error',
+    });
+  }
+};
+
+const checkFileBeforeUpload = (err, req, res, next) => {
+  // FILE SIZE ERROR
+  if (err instanceof multer.MulterError) {
+    return res.status(400).json({
+      status: 'error',
+      code: 400,
+      message: `Max file size ${MAX_AVATAR_FILE_SIZE_IN_BYTES / 1000}KB allowed!`,
+    });
+  }
+
+  // INVALID FILE TYPE, message will return from fileFilter callback
+  else if (err) {
+    return res.status(415).json({ status: 'error', code: 415, message: err.message });
+  }
+  next(req, res, next);
+};
+
+const updateUserAvatar = async (req, res, _) => {
+  try {
+    // FILE NOT SELECTED
+    if (!req.file) {
+      return res.status(400).json({
+        status: 'error',
+        code: 400,
+        message: 'File is required! Please, add the image, if you want to update your avatar.',
+      });
+    }
+
+    const { path: fileFromReqLocatedInAvatarsPath, filename } = req.file;
+    const tempAvatarPath = createFilePath(TMP_DIR, filename);
```

Comment:
dodaj ```.gitkeep``` w folderze tmp bo teraz go nie trzyma w repo

Reply:
nie musi trzymać go w repo, przy uruchomieniu serwera jest funkcja, która sprawdza, czy folder tmp istnieje, jeżeli go nie ma, to go tworzy :) Tak chyba wystarczy? Trzeba dodawać ten plik .gitkeep (ma być pusty w środku?) w tmp?